src/components/Dashboard/DashboardGrid.jsx

```diff
@@ -0,0 +1,173 @@
+ * https://github.com/STRML/react-grid-layout#grid-layout-props
+ *
+ *
+ */
+const DashboardGrid = ({
+  children,
+  breakpoint,
+  isEditable,
+  layouts,
+  onLayoutChange,
+  onBreakpointChange,
+  ...others
+}) => {
+  const generatedLayouts = useMemo(
+    () =>
+      // iterate through each breakpoint
+      Object.keys(DASHBOARD_BREAKPOINTS).reduce((acc, layoutName) => {
+        return {
+          ...acc, // only generate the layout if we're not passed from the parent
+          [layoutName]:
+            layouts && layouts[layoutName]
+              ? layouts[layoutName].map(layout => {
+                  // iterate through all the cards of the laout
+                  // if we can't find the card from the layout, assume small
+                  let matchingCard = find(children, { props: { id: layout.i } });
+                  if (!matchingCard) {
+                    console.warn(`Error with your layout. Card with id: ${layout.i} not found`); //eslint-disable-line
+                    matchingCard = { props: { size: CARD_SIZES.SMALL } };
+                  }
+                  return { ...layout, ...CARD_DIMENSIONS[matchingCard.props.size][layoutName] };
+                })
+              : getLayout(
+                  layoutName,
+                  children.map(card => card.props),
+                  DASHBOARD_COLUMNS,
+                  CARD_DIMENSIONS
+                ),
+        };
+      }, {}),
+    [children, layouts]
+  );
+  const cachedMargin = useMemo(() => [GUTTER, GUTTER], []);
+
+  const handleLayoutChange = useCallback(
+    (layout, allLayouts) => onLayoutChange && onLayoutChange(layout, allLayouts),
+    [onLayoutChange]
+  );
+
+  // add the common measurements and key to the card so that the grid layout can find it
+  const cards = useMemo(
+    () => {
+      return children.map(card =>
+        React.cloneElement(card, {
+          key: card.props.id,
+          dashboardBreakpoints: DASHBOARD_BREAKPOINTS,
+          cardDimensions: CARD_DIMENSIONS,
+          dashboardColumns: DASHBOARD_COLUMNS,
+          rowHeight: ROW_HEIGHT,
+        })
+      );
+    },
+    [children]
+  );
+
+  return (
+    <div style={{ flex: 1 }}>
+      <StyledGridLayout
+        layouts={generatedLayouts}
+        compactType="vertical"
+        cols={DASHBOARD_COLUMNS}
+        breakpoints={DASHBOARD_BREAKPOINTS}
+        margin={cachedMargin}
+        rowHeight={ROW_HEIGHT[breakpoint]}
+        preventCollision={false}
+        // Stop the initial animation unless we need to support editing drag-and-drop
+        shouldAnimate={isEditable}
+        onLayoutChange={handleLayoutChange}
+        onBreakpointChange={onBreakpointChange}
+        isResizable={false}
+        isDraggable={isEditable}
+        {...others}
+      >
+        {cards}
+      </StyledGridLayout>
+    </div>
+  );
+};
+
+DashboardGrid.propTypes = DashboardGridPropTypes;
+DashboardGrid.defaultProps = defaultProps;
+export default DashboardGrid;
```

Comment:
Does this need to be added to the exports in `index.js`?

Reply:
thanks for catching, will fix